services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java

```diff
@@ -72,12 +72,58 @@
     @Autowired
     private SubnetService subnetService;
 
+    @RequestMapping(
+            method = GET,
+            value = {"/subnets/{rangeId}"})
+    @DurationStatistics
+    public Integer getUsedIPByRangeId(@PathVariable String rangeId) throws Exception {
+
+        Integer usedIps = null;
+
+        try {
+            RestPreconditionsUtil.verifyParameterNotNullorEmpty(rangeId);
+
+            usedIps = this.subnetService.getUsedIpByRangeId(rangeId);
+        } catch (ParameterNullOrEmptyException e) {
+            //TODO: REST error code
+            logger.error(e.getMessage());
+            throw new Exception(e);
+        }
```

Comment:
Can we add handling of exceptions other than ParameterNullOrEmptyException?

Reply:
NP

Reply:
For the case of UsedIpsIsNotCorrect, we can directly "throw e" instead of creating a new "empty" exception with the same type.

For the case of ParameterNullOrEmptyException, the new RangeIdIsNullOrEmpty exception doesn't have the actual rangeId. Please include it.

Reply:
Also please remove the TODO comment line as it is actually done.